#!/bin/bash
cd `pwd`
# 1= species 2=reads 3=cp genome
mkdir $1
#mv $1 $3/.
#mv $2 $3/.
#cd $3

gunzip -c $2 > $1/$1_reads.fq
#gunzip -c $2 > $3_reads2.fq
cd $1
source /share/sw/python/2.7.3/.bashrc
time python2.7 /data/kellogg/scripts/cutadapt-1.2.1/bin/cutadapt  -f fastq -b CAAGCAGAAGACGGCATACGAGATCGTGATGTGACTGGAGTTC -b GAACTCCAGTCACATCACGATCTCGTATGCCGTCTTCTGCTTG -b ACACTCTTTCCCTACACGACGCTCTTCCGATCT -b GATCGGAAGAGCGGTTCAGCAGGAATGCCGAG -b AATGATACGGCGACCACCGAGATCTACACTCTTTCCCTACACGACGCTCTTCCGATCT -b CAAGCAGAAGACGGCATACGAGATCGGTCTCGGCATTCCTGCTGAACCGCTCTTCCGATCT -b ACACTCTTTCCCTACACGACGCTCTTCCGATCT -b CGGTCTCGGCATTCCTGCTGAACCGCTCTTCCGATCT -b AGATCGGAAGAGCGTCGTGTAGGGAAAGAGTGT -b CTCGGCATTCCTGCTGAACCGCTCTTCCGATC -b AGATCGGAAGAGCGTCGTGTAGGGAAAGAGTGTAGATCTCGGTGGTCGCCGTATCATT -b AGATCGGAAGAGCGGTTCAGCAGGAATGCCGAGACCGATCTCGTATGCCGTCTTCTGCTTG -b AGATCGGAAGAGCGTCGTGTAGGGAAAGAGTGT -b AGATCGGAAGAGCGGTTCAGCAGGAATGCCGAGACCG -m 40 -e 0.1 -n 3 -O 10 --match-read-wildcards $1_reads.fq > $1.cutadapt

rm $1_reads.fq
#time python2.7 /data/kellogg/scripts/cutadapt-1.2.1/bin/cutadapt  -f fastq -b ACACTCTTTCCCTACACGACGCTCTTCCGATCT -b GATCGGAAGAGCGGTTCAGCAGGAATGCCGAG -b AATGATACGGCGACCACCGAGATCTACACTCTTTCCCTACACGACGCTCTTCCGATCT -b CAAGCAGAAGACGGCATACGAGATCGGTCTCGGCATTCCTGCTGAACCGCTCTTCCGATCT -b ACACTCTTTCCCTACACGACGCTCTTCCGATCT -b CGGTCTCGGCATTCCTGCTGAACCGCTCTTCCGATCT -b AGATCGGAAGAGCGTCGTGTAGGGAAAGAGTGT -b CTCGGCATTCCTGCTGAACCGCTCTTCCGATC -b AGATCGGAAGAGCGTCGTGTAGGGAAAGAGTGTAGATCTCGGTGGTCGCCGTATCATT -b AGATCGGAAGAGCGGTTCAGCAGGAATGCCGAGACCGATCTCGTATGCCGTCTTCTGCTTG -b AGATCGGAAGAGCGTCGTGTAGGGAAAGAGTGT -b AGATCGGAAGAGCGGTTCAGCAGGAATGCCGAGACCG -m 40 -e 0.2 -n 3 -O 10 --match-read-wildcards $2 > $2.cutadapt

#bzip2 $1
#bzip2 $2

time perl /data/kellogg/scripts/new_fastq_cleaner.pl -single $1.cutadapt

rm $1.cutadapt

velveth 71kmer/ 71 -fastq -short $1.cutadapt.cleaned.fq
velvetg 71kmer/

source /share/apps/yasra/.bashrc
cp /share/apps/yasra/test_data/Makefile .
make single_step READS=$1.cutadapt.cleaned.fq TEMPLATE=../$3 TYPE=solexa ORIENT=circular PID=medium

blastall -p blastn -i 71kmer/contigs.fa -o $1_cp.blastn -d ../$3 -m 8 -e 1e-10

perl ../get_blasthit_seqs.pl $1 $1_cp.blastn 71kmer/contigs.fa 
